packages/nuxt/src/common/types.ts

```diff
@@ -128,6 +128,17 @@ export type SentryNuxtModuleOptions = {
    */
   dynamicImportForServerEntry?: boolean;
 
+  /**
+   * The `asyncFunctionReExports` option is only relevant when `dynamicImportForServerEntry: true` (default value).
+   *
+   * As the server entry file is wrapped with a dynamic `import()`, previous async function exports need to be re-exported.
+   * The SDK detects and re-exports those exports (mostly serverless functions). This is why they are re-exported as async functions.
+   * In case you have a custom setup and your server exports other async functions, you can override the default array with this option.
```

Comment:
l: Maybe we can rewrite this a bit, to a perspective of somebody who does not have deep understanding of all of this :D 

```suggestion
   * By default (unless you configure `dynamicImportForServerEntry: false`) the SDK will try to wrap your application entrypoint with a dynamic `import()` to ensure all dependencies can be properly instrumented.
   *
   * By default, the SDK will wrap the default export as well as a `handler` or `server` export from the entrypoint. If your application has a different main export that is used to run the application, you can overwrite this by providing an array of export names to wrap. 
   * Any wrapped export is expected to be an async function.
```